plugins/expired_gpg_keys.py

```diff
@@ -0,0 +1,75 @@
+import dnf
+import rpm
+import subprocess
+from datetime import datetime, timedelta
+from dnfpluginscore import _, logger
+
+
+class ExpiredGPGKeys(dnf.Plugin):
+    """
+    Find expired GPG keys and suggest their removal.
+
+    This is a workaround to solve
+    https://github.com/rpm-software-management/dnf/issues/2075
+    """
+
+    def __init__(self, base, cli):
+        super(ExpiredGPGKeys, self).__init__(base, cli)
+        self.base = base
+        self.cli = cli
+
+    def config(self):
+        """
+        This hook is called immediately after the CLI/extension is finished
+        configuring DNF.
+        """
+        for hdr in self.probably_expired_keys():
+            ctime = datetime.fromtimestamp(hdr[rpm.RPMTAG_INSTALLTIME])
+            nvr = "-".join([hdr["name"], hdr["version"], hdr["release"]])
+
+            print("The following GPG key is probably expired:")
+            print("    {0}\n".format(hdr["summary"]))
+            print("For more information about the key:")
+            print("    rpm -qi {0}\n".format(nvr))
+            print("It was installed at {0}\n".format(ctime))
+
+            print("Trying to install a package signed with this key will "
+                  "result in a failed transaction caused by 'GPG check FAILED'. "
+                  "It is recommended to remove the trusted key so that you are "
+                  "prompted to trust an up-to-date key upon package "
+                  "installation.")
+
+            if self._ask_user_no_raise("Do you want to remove the key?"):
+                cmd = ["rpm", "-e", nvr]
+                result = subprocess.run(cmd)
+                if result.returncode:
+                    print("Err: Failed to remove the '{0}' key".format(nvr))
+                else:
+                    print("Removed: {0}".format(nvr))
+
+    def probably_expired_keys(self):
+        """
+        List expired GPG keys represented as `hdr` objects
+        """
+        ts = rpm.TransactionSet()
+        mi = ts.dbMatch("name", "gpg-pubkey")
+        return [x for x in mi if self.gpgkey_probably_expired(x)]
+
+    def gpgkey_probably_expired(self, hdr):
+        """
+        Is the GPG key expired?
+        """
+        days = 365 * 5 - 30
```

Comment:
This would be much more convenient:
```
$ rpm -q gpg-pubkey-8ddfd64b-65de01d8 --qf '%{DESCRIPTION}' | gpg --show-keys --with-colon | head -1 | cut -d: -f 7
1866728280
$ date --date @1866728280
Ne 25. února 2029, 16:38:00 CET
```
I think that we can expect `gpg` exists everywhere, or simply error out and claiming user should install it.